### PR TITLE
Add springboot test-for-book functionality

### PR DESCRIPTION
**Pull request by AntonZhdanov**

### CODE REVIEW

File: src/main/java/com/example/bookstore/model/Book.java [hunk 1]
```diff
@@ -14,10 +14,14 @@
...
 import lombok.Data;
 import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
 import lombok.ToString;
 import org.hibernate.annotations.SQLDelete;
 import org.hibernate.annotations.Where;
 
+@Getter
+@Setter
```
Comment (keda-vlad) [inline src/main/java/com/example/bookstore/model/Book.java]: @Data contains @Getter and @Setter

File: src/test/java/com/example/bookstore/repository/BookRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        String expectedFieldName = "TITLE";
+
+        // When
+        boolean doesFieldExist = Arrays.stream(BookField.values())
+                .anyMatch(field -> field.name().equals(expectedFieldName));
+
+        // Then
+        assertTrue(doesFieldExist, "Field with name " + expectedFieldName
+                + " does not exist in BookField enum.");
+    }
```
Comment (keda-vlad) [inline src/test/java/com/example/bookstore/repository/BookRepositoryTest.java]: I'm not sure that u need this test, especially in repository test

File: src/test/java/com/example/bookstore/service/CategoryServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        Long categoryId = 1L;
+        Category fantasyCategory = new Category();
+        fantasyCategory.setId(categoryId);
+        fantasyCategory.setName("Fantasy");
+        fantasyCategory.setDescription("Fantasy");
+
+        CategoryDto fantasyCategoryDto = new CategoryDto();
+        fantasyCategoryDto.setId(fantasyCategory.getId());
+        fantasyCategoryDto.setName(fantasyCategory.getName());
+        fantasyCategoryDto.setDescription(fantasyCategory.getDescription());
```
Comment (keda-vlad) [inline src/test/java/com/example/bookstore/service/CategoryServiceImplTest.java]: A lot of repeats, maybe better to make constants or methods

File: src/test/resources/database/books/add-books.sql [hunk 4]
```diff
@@ -0,0 +1,4 @@
+insert into books (id, title, author, isbn, price, description, cover_image, is_deleted) values (1, '1984', 'George Orwell', '978-5-699-12014-7', 19, 'Description', 'Cover image', false);
+insert into books (id, title, author, isbn, price, description, cover_image, is_deleted) values (2, 'The Lord of The Rings', 'J. R. R. Tolkien', 'some isbn2', 29, 'Description', 'Cover image', false);
+insert into books (id, title, author, isbn, price, description, cover_image, is_deleted) values (3, 'The Adventures of Sherlock Holmes', 'Sir Arthur Conan Doyle', 'some isbn3', 24, 'Description', 'Cover image', false);
+insert into books (id, title, author, isbn, price, description, cover_image, is_deleted) values (4, 'A Game of Thrones', ' George RR Martin', 'some isbn4', 34, 'Description', 'Cover image', false);
```
Comment (keda-vlad) [inline src/test/resources/database/books/add-books.sql]: U can make it by one insert
`insert into books (id, title, author, isbn, price, description, cover_image, is_deleted) values 
(1, '1984', 'George Orwell', '978-5-699-12014-7', 19, 'Description', 'Cover image', false), 
(2, 'The Lord of The Rings', 'J. R. R. Tolkien', 'some isbn2', 29, 'Description', 'Cover image', false),
(3, 'The Adventures of Sherlock Holmes', 'Sir Arthur Conan Doyle', 'some isbn3', 24, 'Description', 'Cover image', false),
(4, 'A Game of Thrones', ' George RR Martin', 'some isbn4', 34, 'Description', 'Cover image', false);
`

**Review by keda-vlad — APPROVED** [review]
Approved, but check my comments, it's optional maybe, but maybe will improve yore code)

File: src/test/java/com/example/bookstore/controller/BookControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,474 @@
...
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/books/remove-books.sql"));
+        }
+    }
+
+    @WithMockUser
+    @Test
+    @DisplayName("Get all books")
+    void getAll_ValidPageable_Success() throws Exception {
+        // When
```
Comment (andrii-hoienko) [inline src/test/java/com/example/bookstore/controller/BookControllerTest.java]: Please remove comments

File: src/test/java/com/example/bookstore/controller/BookControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,474 @@
...
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @DisplayName("Update a book with not valid id")
+    void updateBookById_NotValidId_BadRequest() throws Exception {
+        // Given
+        long nonExistingId = 100L;
+
+        CreateBookRequestDto requestDto = new CreateBookRequestDto(
```
Comment (andrii-hoienko) [inline src/test/java/com/example/bookstore/controller/BookControllerTest.java:355]: Please create contents for books and other objects and reuse them

File: src/test/java/com/example/bookstore/controller/BookControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,474 @@
...
+                .andReturn();
+
+        // Then
+        List<BookDto> response = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                new TypeReference<>() {
+                });
+
+        assertFalse(response.isEmpty());
+        assertThat(response).hasSize(4);
```
Comment (andrii-hoienko) [inline src/test/java/com/example/bookstore/controller/BookControllerTest.java:455]: Please make sure that not only size is expected but values as well

File: src/test/java/com/example/bookstore/service/CategoryServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+        // Given
+        when(categoryRepository.findById(anyLong())).thenReturn(Optional.of(fantasyCategory));
+        when(categoryMapper.toDto(any(Category.class))).thenReturn(fantasyCategoryDto);
+
+        // When
+        CategoryDto actual = categoryService.getById(CATEGORY_ID);
+
+        // Then
+        assertEquals(fantasyCategoryDto, actual);
+        verify(categoryRepository, times(1)).findById(anyLong());
```
Comment (andrii-hoienko) [inline src/test/java/com/example/bookstore/service/CategoryServiceImplTest.java]: times(1) is set by default as far as I remember (check all tests)

Comment (andrii-hoienko) [inline src/test/java/com/example/bookstore/service/CategoryServiceImplTest.java]: Not necessary to verify that mocks are called (check all tests)

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Gj! But still some improvements required

File: src/test/java/com/example/bookstore/service/CategoryServiceImplTest.java [hunk 9]
```diff
@@ -93,32 +90,25 @@ void findAll_ValidPageable_ReturnsAllCategories() {
...
         when(categoryMapper.toDto(any(Category.class))).thenReturn(fantasyCategoryDto);
 
-        // When
         CategoryDto actual = categoryService.getById(CATEGORY_ID);
 
-        // Then
         assertEquals(fantasyCategoryDto, actual);
-        verify(categoryRepository, times(1)).findById(anyLong());
-        verifyNoMoreInteractions(categoryRepository, categoryMapper);
+        verify(categoryRepository).findById(anyLong());
```
Comment (vovasalyha) [inline src/test/java/com/example/bookstore/service/CategoryServiceImplTest.java]: you did not resolved Andrii's comment about redundant `verify` method calls, please check the documentation of `verify` method to better understand in which cases it is redundant.
image

**Review by vovasalyha — APPROVED** [review]
Approving, but please check my comment

**Pull request merged**